modules/next_stage/php/next_stage.class.inc

```diff
@@ -119,6 +133,10 @@ class Next_Stage extends \NDB_Form
     function setup()
     {
         parent::setup();
+
+        $this->timepoint = \TimePoint::singleton($this->identifier);
```

Comment:
can you make a redmine ticket to clean this up?

1. I think these are already instantiated and carried with the request in `handle`, so you should be able to just extract these from there (but you should verify that first..)
2. All the other functions in this class are still using singleton instead of using the class variable. They no longer should have any need for that.